library/bignum.c

```diff
@@ -55,9 +55,9 @@
     MBEDTLS_INTERNAL_VALIDATE(cond)
 
 /* Implementation that should never be optimized out by the compiler */
-static void mbedtls_mpi_zeroize(mbedtls_mpi_uint *v, size_t n)
+static void mbedtls_mpi_zeroize_and_free(mbedtls_mpi_uint *v, size_t n)
```

Comment:
Non-blocker: not sure it's worth keeping this function at all.

Reply:
While it does actually provide type safety, it's nice to have `mpi_` in the name, and it shouldn't increase code size at all to do this

Reply:
My concern is more about readability - it's an avoidable step of indirection the reader has to work through. As a quick test, I replaced with a macro and the result saved 8 bytes...

Reply:
Hmm, yes, this function doesn't get inlined. I still prefer having it as a separate function, since it does the `* ciL` scaling for the length.

But yes, this saves another 8 bytes - nice!

Reply:
About making it a macro: do we really need to cater for a missed optimization in one particular compiler jut to save 8 bytes? Does it help if you add `inline`? A macro makes the code harder to follow and there's more risk of getting it wrong (e.g. you almost forgot the parentheses around `n`).